custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js

```diff
@@ -129,47 +129,93 @@ function ServiceDeliveryDashboardController($scope, $http, $location, $routePara
         return renderHeaderTooltip('Growth Monitoring', 'Of the total children between <b>3-5 years</b> of age and enrolled for Anganwadi services, the percentage of children who were weighed in the current month.<br><br><b>Growth Monitoring is done only for children till 5 years of age.</b>');
     }
 
-    function renderPercentageAndPartials(percentage, nominator, denominator) {
-        if (denominator === vm.dataNotEntered) { return vm.dataNotEntered; }
-        if (percentage === vm.dataNotEntered) {
-            if (nominator === 0 && denominator === 0) {
-                return '<div><span>100 %<br>(' + nominator + ' / ' + denominator + ')</span></div>';
+    function isZeroNullUnassignedOrDataNotEntered(value) {
+        return value === 0 || value === null || value === void(0) || value === vm.dataNotEntered;
+    }
+
+    function renderPercentageAndPartials(percentage, nominator, denominator, indicator) {
+        if (isZeroNullUnassignedOrDataNotEntered(denominator) && haveAccessToFeatures) {
+            return '<div> No expected ' + indicator + ' </div>';
+        }
+        else {
+            if (denominator === vm.dataNotEntered) { return vm.dataNotEntered; }
+            if (percentage === vm.dataNotEntered) {
+                if (nominator === 0 && denominator === 0) {
+                    return '<div><span>100 %<br>(' + nominator + ' / ' + denominator + ')</span></div>';
+                }
+                return '<div><span>(' + nominator + ' / ' + denominator + ')</span></div>';
             }
-            return '<div><span>(' + nominator + ' / ' + denominator + ')</span></div>';
         }
+
         return '<div><span>' + percentage + '<br>(' + nominator + ' / ' + denominator + ')</span></div>';
     }
+
+
     function simpleRender(indicator) {
         return function simpleRenderer(data, type, full) {
+
+            if (haveAccessToFeatures && ['state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name', 'num_launched_awcs'].indexOf(indicator) === -1 && isZeroNullUnassignedOrDataNotEntered(full['num_launched_awcs'])) {
+                return '<div>Not Launched</div>';
+            }
+
             return '<div>' + (
                 full[indicator] !== vm.dataNotEntered ? full[indicator] : vm.dataNotEntered
             ) + '</div>';
         };
     }
     function simpleYesNoRender(indicator) {
         return function simpleRenderer(data, type, full) {
-            return '<div>' + (
+
```

Comment:
There is a lot of repeated code here, this should be able to be shared between these calls

Reply:
Done, I have refactored the code  to have minimum replication.
Please review.
@calellowitz